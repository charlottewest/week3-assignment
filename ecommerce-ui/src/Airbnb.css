.airbnb-grid {
  padding: 10px;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 10px;
  grid-auto-rows: repeat(3, 1fr);
  border: 1px solid #DCDCDC;
}

div.airbnbSection {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 10px;
  grid-auto-rows: minmax(100px, auto);
}

img.airbnbImg {
  width: 100%;
  max-width: 400px;
  height: 300px;
  object-fit: fill;
  grid-column: 1 / span 2;
  grid-row: 1;
}

span.airbnbTitle {
  font-size: 15px;
  grid-column: 1 / span 2;
  grid-row: 2;
}

button.airbnbBookingButton {
  font-size: 15px;
  max-width: 80px;
  grid-column: 2;
  grid-row: 3;
  justify-self: end;
}

span.airbnbCost {
  font-size: 15px;
  grid-column: 1;
  grid-row: 3;
}
