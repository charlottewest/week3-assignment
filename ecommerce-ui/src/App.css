div.container {
  padding: 80px;
}

h1.airbnbSectionHeader {
  font-size: 20px;
  text-decoration: underline;
}

div.shoppingCartContainer {
  padding: 50px;
}

div.availablePropertiesContainer {
  padding: 50px;
}
