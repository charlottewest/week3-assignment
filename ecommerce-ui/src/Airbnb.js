import React from 'react';
import './Airbnb.css';

export default class Airbnb extends React.Component {
  render() {
    const {airbnb, onClick, index} = this.props;

    return (
      <div className='airbnb-grid'>
        <img src={this.props.airbnb.image} className='airbnbImg' />
        <span className='airbnbTitle'>{this.props.airbnb.title}</span>
        <span className='airbnbCost'>${this.props.airbnb.payment.cost}</span>
        <button className='airbnbBookingButton' onClick={() => onClick(index)}>Book</button>
      </div>

    )
  }
}
