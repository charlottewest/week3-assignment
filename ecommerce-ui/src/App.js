import React from 'react';
import PropTypes from 'prop-types';
import ShoppingCart from './ShoppingCart';
import Airbnb from './Airbnb';
import AirbnbList from './AirbnbList';
import Airbnbs from './airbnbs.json';
import './App.css';
import './Airbnb.css';

class App extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      airbnbs: Airbnbs,
      selectedAirbnbs: [],
      totalPayment: 0
    }
  }

  bookAirbnb = (index) => {
    const selectedAirbnb = this.state.airbnbs[index];
    if (this.state.selectedAirbnbs.includes(selectedAirbnb)) {
      alert('You already selected that listing.');
      return;
    }

    const newPayment = this.state.totalPayment + selectedAirbnb.payment.cost;

    this.setState(prevState => {
      return {
        selectedAirbnbs: [...prevState.selectedAirbnbs, selectedAirbnb],
        totalPayment: newPayment
      };
    });
  }

  deleteAirbnb = (index) => {
    this.setState(prevState => {
      const selectedAirbnb = this.state.selectedAirbnbs[index];
      const newSelectedAirbnbs = [...this.state.selectedAirbnbs];
      newSelectedAirbnbs.splice(index, 1);
      const newPayment = this.state.totalPayment - selectedAirbnb.payment.cost;

      return {
        selectedAirbnbs: newSelectedAirbnbs,
        totalPayment: newPayment
      };
    });
  }

  render() {
    return (
      <div className='App'>
        <div className='shoppingCartContainer'>
          <h2>Shopping Cart</h2>
          <ShoppingCart
            airbnbs={this.state.selectedAirbnbs}
            ondeleteAirbnb={this.deleteAirbnb}
            totalPayment={this.state.totalPayment}
          />
        </div>
        <hr />
        <div className='availablePropertiesContainer'>
          <h2>Available Properties</h2>
          <AirbnbList
            airbnbs={this.state.airbnbs}
            onbookAirbnb={this.bookAirbnb}
          />
        </div>
      </div>
    );
  }
}

export default App;
