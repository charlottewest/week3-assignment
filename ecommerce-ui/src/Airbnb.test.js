import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, render, mount} from 'enzyme';
import App from './App';
import Airbnb from './Airbnb';
import Airbnbs from './airbnbs.json';

it('shallow renders', () => {
  const testAirbnb = Airbnbs[0]
  const wrapper = shallow(<Airbnb airbnb={testAirbnb}/>);
  expect(wrapper.render()).toMatchSnapshot();

  expect(wrapper.props('children')).toBeDefined();
});
