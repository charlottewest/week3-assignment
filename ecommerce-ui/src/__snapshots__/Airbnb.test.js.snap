// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders 1`] = `
<div
  class="airbnb-grid"
>
  <img
    class="airbnbImg"
    src="https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80"
  />
  <span
    class="airbnbTitle"
  >
    Centrally-located Manhattan studio
  </span>
  <span
    class="airbnbCost"
  >
    $105
  </span>
  <button
    class="airbnbBookingButton"
  >
    Book
  </button>
</div>
`;
